src/netlify/deploy/index.ts

```diff
@@ -37,31 +32,25 @@ type RETVAL = {
  *
  * @returns {object} The Netlify deployment and log URLs.
  */
-export const execute = async (args: INetlifyDeployArgs): Promise<RETVAL | undefined> => {
-  let retVal: RETVAL | undefined;
-
-  try {
-    const token = args.token || (await getToken(args.spinner));
-    const client = new NetlifyAPI(token);
-    const siteId = args.siteId || (await getSiteId(args.spinner));
-
-    /* https://open-api.netlify.com/#operation/createSiteDeploy */
-    const response = await client.deploy(siteId, args.dir, {
-      draft: !args.production,
-      message: args.message
-    });
-
-    const url = args.production ? response.deploy.ssl_url : response.deploy.deploy_ssl_url;
-    const logUrl = `${response.deploy.admin_url}/deploys/${response.deploy.id}`;
+export const execute = async (args: INetlifyDeployArgs): Promise<string> => {
+  const token = args.token || (await getToken(args.spinner));
+  const siteId = args.siteId || (await getSiteId(args.spinner));
+  const production = args.production ? '--prod' : '';
 
-    retVal = { url, logUrl };
-  } catch (error: unknown) {
-    handleErrorMessage(error, 'netlify-deploy', args.spinner);
-
-    throw error;
-  }
-
-  return retVal;
+  return new Promise((resolve, reject) => {
+    exec(
+      `netlify deploy ${production} --dir=${args.dir} --site=${siteId} --auth=${token} --message=${args.message}`,
```

Comment:
I've verified deployments locally. However, the `netlify` command will need to be available in CI environments. Netlify recommends installing `netlify-cli` as a local dependency per project. See ["Installing in a CI environment"](https://docs.netlify.com/cli/get-started/#installation-in-a-ci-environment).